CMakelists.txt

```diff
@@ -0,0 +1,124 @@
+cmake_minimum_required(VERSION 3.0)
+
+project(rabit VERSION 1.0.0)
+
+option(RABIT_BUILD_TESTS "Build rabit tests" OFF)
+option(RABIT_BUILD_MPI "Build MPI" OFF)
+
+add_library(rabit src/allreduce_base.cc src/allreduce_robust.cc src/engine.cc src/c_api.cc)
+add_library(rabit_base src/allreduce_base.cc src/engine_base.cc src/c_api.cc)
+add_library(rabit_empty src/engine_empty.cc src/c_api.cc)
+
+set(rabit_libs rabit rabit_base rabit_empty)
+if(RABIT_BUILD_MPI)
+  add_library(rabit_mpi src/engine_mpi.cc)
+  list(APPEND rabit_libs rabit_mpi)
+endif()
+
+if(RABIT_BUILD_TESTS)
+  add_library(rabit_mock src/allreduce_base.cc src/allreduce_robust.cc src/engine_mock.cc src/c_api.cc)
+  list(APPEND rabit_libs rabit_mock) # add to list to apply build settings, then remove
+endif()
+
+foreach(lib ${rabit_libs})
+  #include "./internal/utils.h"  
+  target_include_directories(${lib} PUBLIC
+    "$<INSTALL_INTERFACE:include>"    
```

Comment:
Not needed, already set by `install(... INCLUDES DESTINATION ...)`

Reply:
done